src/main/java/org/elasticsearch/common/io/PathUtils.java

```diff
@@ -75,7 +75,22 @@ public static Path get(URI uri) {
             return Paths.get(uri);
         }
     }
-    
+
+    /**
+     * Tries to resolve the given path against the list of available roots.
+     *
+     * If path starts with one of the listed roots, it returned back by this method, otherwise null is returned.
+     */
+    public static Path get(Path[] roots, String path) {
+        Path normalizedPath = DEFAULT.getPath(path).normalize();
+        for (Path root : roots) {
+            if(normalizedPath.startsWith(root)) {
+                return normalizedPath;
+            }
+        }
```

Comment:
@rmuir so in this case if a relative path is specified for location it would be resolved against the first root. I think I like it.